renderdoc/driver/gl/gl_counters.cpp

```diff
@@ -274,7 +273,7 @@ void GLReplay::FillTimers(GLCounterContext &ctx, const DrawcallDescription &draw
 
     m_pDriver->ReplayLog(ctx.eventStart, d.eventId, eReplay_OnlyDraw);
 
-    for(auto q : indices<GPUCounter>())
+    for(uint32_t q = 0; q < uint32_t(GPUCounter::Count); q++)
```

Comment:
I also don't understand this change for similar reasons to above. the `indices<GPUCounter>()` is just shorthand for iterating from `GPUCounter::First` to `GPUCounter::Count-1`. The loop only changes by including `q==0` which should never correspond to a valid counter.

Where was this necessary? When was a counter with `q==0` begun? This suggests a more serious issue and this change is hiding the problem.

Reply:
Oops, yeah 0 is invalid here...
I just run into problems of unended queries and that seem to fix it.
Putting a log trace into this for loop never printed anything, even though the preceding loop did...
Where is this indices template define?

Reply:
The template is defined in `renderdoc_replay.h` but I don't think there is a problem with the template, as the for loop works fine for me, if I add a log then I get prints from `q==1` to `q==13`.

Can you give more information about your problem and how to reproduce it? Does it reproduce on a simple example like `https://github.com/gears3d/gears3d`?